apps/cancel-request/fields/index.js

```diff
@@ -2,6 +2,14 @@
 const dateComponent = require('hof').components.date;
 const countries = require('hof').utils.countries();
 
+function hoRefNum(value) {
+  return value.match(/^[A-Z]\d+$/i);
+}
+
+function recordNum(value) {
+  return value.match(/^(ROD\d{9}|\d{9})$/i);
+}
```

Comment:
Regex should allow 'r0d' too

If you enter 'rod' or 'r0d' it should allow 9-12 digits I don't think this will work though if you have a maxlength validator from hof so i think it'll need some custom validation similar to lcs>custom-validation.js?

Reply:
This is case incentive match. So, it will allow rOd. My understanding is that, if user enters ROD then we allow 9 digits else we just accept 6 digits (without the keyword rod). isnt it?

Reply:
Regex is now updated to accept 0 as well i.e. R0D or r0d, etc.